$OpenBSD: patch-src_rrd_daemon_c,v 1.1 2015/11/23 11:41:02 sthen Exp $

From 2a1731043212a8fa623566f7550c052f9c031860 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Michel=20Vourg=C3=A8re?= <nirgal@debian.org>
Date: Wed, 26 Aug 2015 12:10:59 +0200
Subject: [PATCH] Fixed compilation when there's no sigwaitinfo

* Fixed pointer in call to sigwait
* Don't print pid when we don't have it
* Fixed unused variable warning

--- src/rrd_daemon.c.orig	Tue Nov 10 17:15:18 2015
+++ src/rrd_daemon.c	Mon Nov 23 11:26:04 2015
@@ -335,16 +335,16 @@ static void sig_common (const char *sig) /* {{{ */
 
 static void* signal_receiver (void UNUSED(*args))
 {
-  siginfo_t signal_info;
   int status;
 
   while (1)
   {
 #if defined(HAVE_SIGWAITINFO)
+    siginfo_t signal_info;
     status = sigwaitinfo(&signal_set, &signal_info);
 #elif defined(HAVE_SIGWAIT)
     status = -1;
-    if (sigwait(signal_set, &status) < 0 ){
+    if (sigwait(&signal_set, &status) < 0 ){
        status = -1;
     }
 #else
@@ -408,11 +408,18 @@ static void* signal_receiver (void UNUSED(*args))
         break;
 
       default:
+#if defined(HAVE_SIGWAITINFO)
         RRDD_LOG(LOG_NOTICE,
                  "%s: Signal %d was received from process %u.\n",
                  __func__,
                  status,
                  signal_info.si_pid);
+#else
+        RRDD_LOG(LOG_NOTICE,
+                 "%s: Signal %d was received.\n",
+                 __func__,
+                 status);
+#endif
     }
   }
 
